network/src/peer_store/db.rs

```diff
@@ -0,0 +1,313 @@
+use super::{Multiaddr, PeerId, Score, Status};
+use crate::network_group::{Group, NetworkGroup};
+use libp2p::core::Endpoint;
+use rusqlite::types::ToSql;
+use rusqlite::{Connection, NO_PARAMS};
+use std::time::Duration;
+
+#[derive(Debug)]
+pub struct PeerInfo {
+    pub id: u32,
+    pub peer_id: PeerId,
+    pub connected_addr: Multiaddr,
+    pub score: Score,
+    pub status: Status,
+    pub endpoint: Endpoint,
+    pub ban_time: Duration,
+    pub connected_time: Duration,
+}
+
+pub fn create_tables(conn: &Connection) {
+    let sql = r#"
+    CREATE TABLE IF NOT EXISTS peer_info (
+    id INTEGER PRIMARY KEY NOT NULL,
+    peer_id BINARY UNIQUE NOT NULL,
```

Comment:
need to add index for `peer_id`

Reply:
The UNIQUE keyword already implied index this column which is a convention in relation DBs.